internal/database/release.go

```diff
@@ -447,3 +447,66 @@ func (repo *ReleaseRepo) Delete(ctx context.Context) error {
 
 	return nil
 }
+
+func (repo *ReleaseRepo) CanDownloadShow(ctx context.Context, title string, season int, episode int) (bool, error) {
+	// TODO support non season episode shows
+	// if rls.Day > 0 {
+	//	// Maybe in the future
+	//	// SELECT '' FROM release WHERE Title LIKE %q AND ((Year == %d AND Month == %d AND Day > %d) OR (Year == %d AND Month > %d) OR (Year > %d))"
+	//	qs := sql.Query("SELECT torrent_name FROM release WHERE Title LIKE %q AND Year >= %d", rls.Title, rls.Year)
+	//
+	//	for q := range qs.Rows() {
+	//		r := rls.ParseTitle(q)
+	//		if r.Year > rls.Year {
+	//			return false, fmt.Errorf("stale release year")
+	//		}
+	//
+	//		if r.Month > rls.Month {
+	//			return false, fmt.Errorf("stale release month")
+	//		}
+	//
+	//		if r.Month == rls.Month && r.Day > rls.Day {
+	//			return false, fmt.Errorf("stale release day")
+	//		}
+	//	}
+	//}
+
+	queryBuilder := repo.db.squirrel.
+		Select("COUNT(*)").
+		From("release").
+		Where("title LIKE ?", fmt.Sprint("%", title, "%"))
+
+	if season > 0 && episode > 0 {
```

Comment:
These aren't quite right, they need to check the present season and episodes against that, or future seasons registered. Right now it's split brain between packs and episodes, as well as episodes failing on future seasons.

Reply:
Took a stab at it.